File: SEP3_UserSystem/src/main/java/model/domain/List/userList/AccountList.java
package model.domain.list.userList;

import model.domain.unit.user.Account;

import java.util.ArrayList;

public class AccountList {
    private ArrayList<Account> Accounts;

    public AccountList()
    {
        Accounts = new ArrayList<>();
    }

    public String addNewAccount(Account newAccount)
    {
        if (getAccountById(newAccount.getId())!=null)
        {
            return "Account [" + newAccount.getId() + "] is not a new account.";
        }
        Accounts.add(newAccount);
        return null;
    }

    public Account getAccountById(String id)
    {
        for (int x = 0; x< Accounts.size(); x++)
        {
            if (Accounts.get(x).getId().equals(id))
            {
                return Accounts.get(x).copy();
            }
        }
        return null;
    }

    public String changePassword(String id,String oldPassword,String newPassword)
    {
        for (int x = 0; x< Accounts.size(); x++)
        {
            if (Accounts.get(x).getId().equals(id))
            {
                return Accounts.get(x).changePassWord(oldPassword, newPassword);
            }
        }
        return "Can't find user [" + id + "].";
    }

    public String updateAccount(Account account)
    {
        for (int x = 0; x< Accounts.size(); x++)
        {
            if (Accounts.get(x).getId().equals(account.getId()))
            {
                Accounts.get(x).update(account);
                return null;
            }
        }
        return "Can't find account [" + account.getId() + "].";
    }

    public void removeAccountById(String id)
    {
        for (int x = 0; x< Accounts.size(); x++)
        {
            if (Accounts.get(x).getId().equals(id))
            {
                Accounts.remove(x);
                break;
            }
        }
    }

    public AccountList copy()
    {
        AccountList copy = new AccountList();
        for (int x = 0; x< Accounts.size(); x++)
        {
            copy.addNewAccount(Accounts.get(x).copy());
        }
        return copy;
    }
}

File: SEP3_Client/Model/List/UserList/AccountList.cs
using System;
using System.Collections.Generic;
using SEP3_Client.Model.Unit.User;

namespace SEP3_Client.Model.List.UserList
{
    public class AccountList
    {
        public List<Account> Accounts { get; set; }

        public AccountList()
        {
            Accounts = new List<Account>();
        }

        public string AddNewAccount(Account newAccount)
        {
            if (GetAccountById(newAccount.GetId())!=null)
            {
                return "Account [" + newAccount.GetId() + "] is not a new account.";
            }
            Accounts.Add(newAccount);
            return null;
        }
        
        public string AddOldAndNewAccount(Account oldAccount,Account newAccount)
        {
            Accounts = new List<Account> {oldAccount, newAccount};
            return null;
        }
        
        public Account GetAccountByIndex(int index)
        {
            if (index>=0&&index<Accounts.Count)
            {
                return Accounts[index];
            }
            else
            {
                return null;
            }
        }

        public Account GetAccountById(string id)
        {
            foreach (var account in Accounts)
            {
                if (account.GetId().Equals(id))
                {
                    return account;
                }
            }
            return null;
        }

        public string ChangePassword(string id,string oldPassword,string newPassword)
        {
            foreach (var account in Accounts)
            {
                if (account.GetId().Equals(id))
                {
                    return account.ChangePassWord(oldPassword, newPassword);
                }
            }

            return "Can't find user [" + id + "].";
        }

        public string UpdateAccount(Account newAccount)
        {
            foreach (var account in Accounts)
            {
                if (account.GetId().Equals(newAccount.GetId()))
                {
                    account.Update(newAccount);
                    return null;
                }
            }

            return "Can't find account [" + newAccount.GetId() + "].";
        }

        public void RemoveAccountById(string id)
        {
            foreach (var account in Accounts)
            {
                if (account.GetId().Equals(id))
                {
                    Accounts.Remove(account);
                    break;
                }
            }
        }

        public AccountList Copy()
        {
            AccountList copy = new AccountList();
            foreach (var account in Accounts)
            {
                copy.AddNewAccount(account.Copy());
            }
            return copy;
        }
    }
}